### ISSUE TITLE
Getting device info out of the webservice necessary ?

### ISSUE THREAD

**Issue by mt7479** [AndersRodland/ConfigMgrClientHealthWebservice#1]
Without any kind of authentication it's currently possible to:

* obtain individual device info by accessing /clients/hostname
* obtain a complete json dump by accessing /clients

if the URL is known to an interested party. It would be nice to have config options to disable this behavior and/or limit information to a minimum.

Today this can be remediated by leveraging the IIS rewrite module with something like this:

```
<rewrite>
	<rules>
		<rule name="deny-client-dump" enabled="true" stopProcessing="true">
			<match url=".*" />
			<conditions logicalGrouping="MatchAll">
				<add input="{URL}" pattern="\/clients\/?$" />
				<add input="{REQUEST_METHOD}" pattern="PUT|POST" negate="true" />
			</conditions>
			<action type="CustomResponse" statusCode="403" statusReason="Forbidden: Access is denied." statusDescription="You do not have permission to view this directory or page using the credentials that you supplied." />
		</rule>
	</rules>
	<outboundRules>
		<rule name="deny-specific-client-result" preCondition="match-content" enabled="true">
			<match filterByTags="None" pattern=".*" />
			<conditions logicalGrouping="MatchAll">
				<add input="{RESPONSE_STATUS}" pattern="200" />
			</conditions>
			<action type="Rewrite" value="" />
		</rule>
		<preConditions>
			<preCondition name="match-content">
				<add input="{RESPONSE_CONTENT_TYPE}" pattern="application/json" />
			</preCondition>
		</preConditions>
	</outboundRules>
</rewrite>
```
This will:

* allow only PUT & POST on /clients /clients/ and return 403 on everything else
* return an empty response on /clients/hostname if content type is json and success code 200

The outgoing response rule is enough to send empty json data, but blocking incoming /get requests on /clients has a performance benefit.

**Comment by AndersRodland** [comment]
You are right, this is not necessary, and this part of the code should never had made it to production. I will remove it in the next version.

**Comment by mt7479** [comment]
Thanks for the reply, looking forward to the next release.

**Comment by AndersRodland** [comment]
Finally implemented in version 2.0
